app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -1975,52 +1972,54 @@ class BrowserTabFragment :
         )
     }
 
-    private fun askSiteLocationPermission(locationPermission: LocationPermission) {
+    private fun askSiteLocationPermission(locationPermissionRequest: LocationPermissionRequest) {
         if (!isActiveCustomTab() && !isActiveTab) {
             Timber.v("Will not launch a dialog for an inactive tab")
             return
         }
 
-        val binding = ContentSiteLocationPermissionDialogBinding.inflate(layoutInflater)
-
-        val domain = locationPermission.origin
-        val title = domain.websiteFromGeoLocationsApiOrigin()
-        binding.sitePermissionDialogTitle.text = getString(R.string.preciseLocationSiteDialogTitle, title)
-        binding.sitePermissionDialogSubtitle.text = if (title == DDG_DOMAIN) {
-            getString(R.string.preciseLocationDDGDialogSubtitle)
-        } else {
-            getString(R.string.preciseLocationSiteDialogSubtitle)
-        }
-
-        val dialog = MaterialAlertDialogBuilder(requireActivity())
-            .setView(binding.root)
-            .setOnCancelListener {
-                // Called when user clicks outside the dialog - deny to be safe
-                locationPermission.callback.invoke(locationPermission.origin, false, false)
-            }
-            .create()
-
-        binding.siteAllowAlwaysLocationPermission.setOnClickListener {
-            viewModel.onSiteLocationPermissionSelected(domain, LocationPermissionType.ALLOW_ALWAYS)
-            dialog.dismiss()
-        }
-
-        binding.siteAllowOnceLocationPermission.setOnClickListener {
-            viewModel.onSiteLocationPermissionSelected(domain, LocationPermissionType.ALLOW_ONCE)
-            dialog.dismiss()
-        }
-
-        binding.siteDenyOnceLocationPermission.setOnClickListener {
-            viewModel.onSiteLocationPermissionSelected(domain, LocationPermissionType.DENY_ONCE)
-            dialog.dismiss()
-        }
-
-        binding.siteDenyAlwaysLocationPermission.setOnClickListener {
-            viewModel.onSiteLocationPermissionSelected(domain, LocationPermissionType.DENY_ALWAYS)
-            dialog.dismiss()
-        }
-
-        dialog.show()
+        activity?.let {
+            sitePermissionsDialogLauncher.showSiteLocationPermissionDialog(it, locationPermissionRequest, tabId)
+        }
+
+        // val binding = ContentSiteLocationPermissionDialogBinding.inflate(layoutInflater)
```

Comment:
Note to remove commented out code (I see it was moved).

Reply:
It’s removed a bit higher, alongside the resources, etc